tests/main/lxd/prep-snapd-in-lxd.sh

```diff
@@ -9,6 +9,24 @@ if [ -e /var/lib/dpkg/info/snapd.postrm ]; then
     sed -i 's#echo "Final directory cleanup"#umount /snap || true#' /var/lib/dpkg/info/snapd.postrm
 fi
 
+# wait until we can do a DNS lookup on ubuntu.com before doing any apt 
+# operations, as the lxd container sometimes sees really weird errors about the 
+# apt sources.list such as:
+#
+# Reading package lists...
+# E: Type 'ubuntu' is not known on line 50 in source list /etc/apt/sources.list
```

Comment:
This is slightly puzzling. The error there comes from apt-pkg/sourceslist.cc:ParseFileOldStyle() and it indicates that sources.list has a line that starts with "ubuntu", e.g.:
```
$ tail -n1 /etc/apt/sources.list
ubuntu	http://archive.ubuntu.com/ubuntu/ groovy main restricted
$ sudo apt update
E: Type 'ubuntu' is not known on line 61 in source list /etc/apt/sources.list
E: The list of sources could not be read.
```
So the fact that this goes away when dns is around is interesting. I suspect it's a race somehow, i.e. a template sources.list is written by something and then the template is expanded. Hence the wait fixes it. Maybe @stgraber knows more how lxd handles this?

Reply:
it is apparently cloud-init which re-writes the apt sources file, so I changed this to wait for cloud-init to be done before trying any `apt` commands. DNS would not work until after cloud-init is done configuring the network so that's probably why it worked to wait until DNS was done (because we were effectively waiting for cloud-init to be done).